paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -176,17 +212,45 @@ def get_services_to_k8s_namespaces(
                     soa_dir=soa_dir,
                 )
                 if instances:
-                    services_to_k8s_namespaces[service].add(
+                    # Currently, all instance types besides kubernetes use one big namespace, defined in
+                    # INSTANCE_TYPE_TO_K8S_NAMESPACE. Sync all shared secrets and all secrets belonging to any service
+                    # which uses that instance type.
+
+                    services_to_k8s_namespaces_to_allowlist[service][
                         INSTANCE_TYPE_TO_K8S_NAMESPACE[instance_type]
-                    )
+                    ] = None
+                    if "_shared" in service_list:
+                        services_to_k8s_namespaces_to_allowlist["_shared"][
+                            INSTANCE_TYPE_TO_K8S_NAMESPACE[instance_type]
+                        ] = None
+
+    return dict(services_to_k8s_namespaces_to_allowlist)
+
+
+def get_secrets_used_by_instance(
+    service_instance_config: KubernetesDeploymentConfig,
+) -> Tuple[Set[str], Set[str]]:
```

Comment:
might be nice to turn this into a typeddict/namedtuple/dataclass so that we can handle other secret types more easily in the future

Reply:
(although, I guess that maybe we just stuff any other secret types into `secrets_used` and not worry about env var vs non-env var secrets :p)